### PR TITLE
docs: document hasMany/belongsTo/hasOne limitations for NoSQL databases

### PR DESCRIPTION
**Pull request by hacksparrow**
Documented hasMany/belongsTo/hasOne limitations for NoSQL databases.

Addresses https://github.com/strongloop/loopback-next/issues/2340.

### CODE REVIEW

File: docs/site/BelongsTo-relation.md [hunk 1]
```diff
@@ -8,6 +8,14 @@ permalink: /doc/en/lb4/BelongsTo-relation.html
 
 ## Overview
 
+{% include note.html content="
+This relation best works with databases that support foreign key
+constraint (SQL).
+Using this relation with NoSQL databases will result in unexpected behavior,
+such as the ability to create a relation with a model that does not exist. We are working on a solution to better handle this. It is fine to use this relation with NoSQL databases for purposes such as navigating
+related models, where the referential integrity as not critical.
```
Comment (raymondfeng) [inline docs/site/BelongsTo-relation.md]: `as not critical` -> `is not critical`.

File: docs/site/BelongsTo-relation.md [hunk 2]
```diff
@@ -8,6 +8,14 @@ permalink: /doc/en/lb4/BelongsTo-relation.html
 
 ## Overview
 
+{% include note.html content="
+This relation best works with databases that support foreign key
+constraint (SQL).
```
Comment (raymondfeng) [inline docs/site/BelongsTo-relation.md]: constraint -> constraints?

File: docs/site/BelongsTo-relation.md [hunk 3]
```diff
@@ -8,6 +8,14 @@ permalink: /doc/en/lb4/BelongsTo-relation.html
...
+Using this relation with NoSQL databases will result in unexpected behavior,
+such as the ability to create a relation with a model that does not exist. We are working on a solution to better handle this. It is fine to use this relation with NoSQL databases for purposes such as navigating
```
Comment (nabdelgadir) [inline docs/site/BelongsTo-relation.md]: `We are working on a solution to better handle this` is there a GitHub issue we can link them to?

Comment (hacksparrow) [reply docs/site/BelongsTo-relation.md]: AFAIK, there is none. An issue was opened to initiate the discussion on the matter - https://github.com/strongloop/loopback-next/issues/2127 - it is closed now. @bajtos, you thougts?

**Comment by hacksparrow** [comment]
Thanks @raymondfeng, I have updated the files to address the issues pointed out by you.

File: docs/site/BelongsTo-relation.md (hunk 3 above)
Comment (bajtos) [reply docs/site/BelongsTo-relation.md]: Let's use the following issue for the link: _Docs: what relation types will work well with my database?_ #2341

File: docs/site/Relations.md [hunk 4]
```diff
@@ -39,6 +39,14 @@ Here are the currently supported relations:
 - [BelongsTo](BelongsTo-relation.md)
 - [HasOne](hasOne-relation.md)
 
+{% include note.html content="
+The `hasMany` relation may alternatively be implemented using the
+`referencesMany` and `embedsMany` relations. These relations are similar, but
+not the same. Since each database paradigm comes with different trade-offs and
+thus different databases require the applications to use different relation
+types, use the relation best suited for your database.
```
Comment (bajtos) [inline docs/site/Relations.md]: Please add a link to https://github.com/strongloop/loopback-next/issues/2341

**Comment by hacksparrow** [comment]
Thanks @bajtos, I have added the links.

File: docs/site/HasMany-relation.md [hunk 5]
```diff
@@ -8,6 +8,14 @@ permalink: /doc/en/lb4/HasMany-relation.html
 
 ## Overview
 
+{% include note.html content="
+This relation best works with databases that support foreign key
+constraints (SQL).
+Using this relation with NoSQL databases will result in unexpected behavior,
+such as the ability to create a relation with a model that does not exist. We are working on a solution to better handle this. It is fine to use this relation with NoSQL databases for purposes such as navigating
```
Comment (bajtos) [inline docs/site/HasMany-relation.md]: Please add the link here too.

File: docs/site/Relations.md [hunk 6]
```diff
@@ -39,6 +39,14 @@ Here are the currently supported relations:
 - [BelongsTo](BelongsTo-relation.md)
 - [HasOne](hasOne-relation.md)
 
+{% include note.html content="
+The `hasMany` relation may alternatively be implemented using the
+`referencesMany` and `embedsMany` relations. These relations are similar, but
+not the same. Since each database paradigm comes with different trade-offs and
+thus different databases require the applications to use different relation
+types, use the relation best [suited for your database](https://github.com/strongloop/loopback-next/issues/2341).
```
Comment (bajtos) [inline docs/site/Relations.md]:
```suggestion
types, use the [relation best suited for your database](https://github.com/strongloop/loopback-next/issues/2341).
```

File: docs/site/hasOne-relation.md [hunk 7]
```diff
@@ -8,6 +8,14 @@ permalink: /doc/en/lb4/hasOne-relation.html
 
 ## Overview
 
+{% include note.html content="
+This relation best works with databases that support foreign key and unique
+constraints (SQL).
+Using this relation with NoSQL databases will result in unexpected behavior,
+such as the ability to create a relation with a model that does not exist. We are working on a solution to better handle this. It is fine to use this relation with NoSQL databases for purposes such as navigating
```
Comment (bajtos) [inline docs/site/hasOne-relation.md]: Please add the link here too.

**Pull request merged**